src/lua/ext_string.lua

```diff
@@ -0,0 +1,288 @@
+local ffi = require('ffi')
+
+ffi.cdef[[
+    const char *
+    memmem(const char *haystack, size_t haystack_len,
+           const char *needle,   size_t needle_len);
+    int memcmp(const char *mem1, const char *mem2, size_t num);
+    int isspace(int c);
+]]
+
+local c_char_ptr = ffi.typeof('const char *')
+
+local memcmp  = ffi.C.memcmp
+local memmem  = ffi.C.memmem
+local isspace = ffi.C.isspace
+
+local err_string_arg = "bad argument #%d to '%s' (%s expected, got %s)"
+
+local function string_split_empty(inp, maxsplit)
+    local p = c_char_ptr(inp)
+    local p_end = p + #inp
+    local rv = {}
+    while true do
+        -- skip the leading whitespaces
+        while p < p_end and isspace(p[0]) ~= 0 do
+            p = p + 1
+        end
+        if p == p_end then
+            break
+        end
+        if maxsplit <= 0 then
+            table.insert(rv, ffi.string(p, p_end - p))
+            break
+        end
+        local chunk = p
+        -- skip all non-whitespace characters
+        while p < p_end and isspace(p[0]) == 0 do
+            p = p + 1
+        end
+        assert((p - chunk) > 0)
+        table.insert(rv, ffi.string(chunk, p - chunk))
+        maxsplit = maxsplit - 1
+    end
+    return rv
+end
+
+local function string_split_internal(inp, sep, maxsplit)
+    local p = c_char_ptr(inp)
+    local p_end = p + #inp
+    local sep_len = #sep
+    if sep_len == 0 then
+        error(err_string_arg:format(2, 'string.split', 'non-empty string',
+              "empty string"), 3)
+    end
+    local rv = {}
+    while true do
+        assert(p <= p_end)
+        if maxsplit <= 0 or p == p_end then
+            table.insert(rv, ffi.string(p, p_end - p))
+            break
+        end
+        local chunk = p
+        p = memmem(p, p_end - p, sep, sep_len) or p_end
+        if p == nil then
```

Comment:
"or p_end" is an artifact. It didn't work as expected and I had to add the if above:

```
tarantool> NULL  = ffi.cast('void *', 0)
---
...

tarantool> NULL or 48
---
- null
...

tarantool> nil or 48
---
- 48
...
```

I'll remove "or p_end", thanks!